### PR TITLE
TRUNK-5868: OrderType Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-5868
## Checklist: I completed these to help reviewers :)
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.
- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by alinmihaila99** [comment]
Hi @Ruhanga, I made the changes you suggested and still got that error.
https://pastebin.com/m1TczrcP

**Comment by alinmihaila99** [comment]
@Ruhanga , I pushed the changes

**Comment by gitcliff** [comment]
@alinmihaila99  i see the cause of build failure could be `Caused by: org.dbunit.dataset.NoPrimaryKeyException: ORDER_TYPE_CLASS_MAP`

**Comment by alinmihaila99** [comment]
@gitcliff I know, I put that error in the pastebin. But I don't know how to solve it

File: api/src/main/java/org/openmrs/OrderType.java [hunk 1]
```diff
@@ -30,13 +50,24 @@
...
+	@ManyToOne
+	@JoinColumn(name = "parent")
 	private OrderType parent;
 	
 	@Independent
+	@ManyToMany
+	@LazyCollection(LazyCollectionOption.TRUE)
+	@JoinTable(name = "order_type_class_map", joinColumns = @JoinColumn(name = "order_type_id", updatable = false, insertable = false), 
+		inverseJoinColumns = @JoinColumn(name = "concept_class_id", unique = true))
 	private Collection<ConceptClass> conceptClasses;
```
Comment (Ruhanga) [inline api/src/main/java/org/openmrs/OrderType.java:71]: @alinmihaila99, you could probably transform and specify `conceptClasses` into a `Set` and provide the getter/setter in a backwards compatible way. This means you'd maintain the old setter/getter methods but add some casting to them so implementations specifically relying on these methods in a tightly coupled way might still use them. 

Do you have any thoughts or preference on this @ibacher, @dkayiwa?

**Review by Ruhanga — APPROVED** [review]

**Pull request closed without merging**